tests/L0/run_instance_norm_nvfuser/test_instance_norm_nvfuser.py

```diff
@@ -0,0 +1,69 @@
+
+class TestInstanceNormNVFuser(unittest.TestCase):
+    dtype = torch.float
+    track_running_stats = False
+    channels_last = False
+    affine = False
+    batch_size = 5
+    channel_size = 7
+    spatial_size = 3
+
+    def setUp(self):
+        self.m = InstanceNorm3dNVFuser(self.channel_size, affine=self.affine, track_running_stats=self.track_running_stats, device='cuda', dtype=self.dtype)
+        self.reference_m = torch.nn.InstanceNorm3d(self.channel_size, affine=self.affine, track_running_stats=self.track_running_stats, device='cuda', dtype=self.dtype)
+
+    def check_same_output(self):
+        torch.manual_seed(42) 
+        for i in range(2): # exercise JIT + caching
+            inp = torch.randint(0, 2, (self.batch_size, self.channel_size, self.spatial_size, self.spatial_size, self.spatial_size), device='cuda', requires_grad=True, dtype=self.dtype)
+            inp2 = inp.detach().clone()
+            inp2.requires_grad = True
+            if self.channels_last:
+                _inp = inp.to(memory_format=torch.channels_last_3d)
+            else:
+                _inp = inp
+            out = self.m(_inp)
+            (out.sum()).backward()
+            out2 = self.reference_m(inp2)
+            if self.m.running_mean is None:
+                assert self.reference_m.running_mean is None
+                assert self.m.running_var is None
+                assert self.reference_m.running_var is None
+            else:
+                torch.testing.assert_close(self.m.running_mean, self.reference_m.running_mean)
+                if self.dtype == torch.float16:
+                    torch.testing.assert_close(self.m.running_var, self.reference_m.running_var, atol=5e-3, rtol=5e-3)
+                else:
+                    torch.testing.assert_close(self.m.running_var, self.reference_m.running_var)
+            torch.testing.assert_close(out, out2)
+            (out2.sum()).backward()
+            if self.dtype == torch.float16:
+                torch.testing.assert_close(inp.grad, inp2.grad, atol=5e-3, rtol=5e-3)
+            else:
+                torch.testing.assert_close(inp.grad, inp2.grad)
+            if self.m.weight is not None:
+                if self.dtype == torch.float16:
+                    torch.testing.assert_close(self.m.weight.grad, self.reference_m.weight.grad, atol=5e-2, rtol=5e-2)
+                else:
+                    torch.testing.assert_close(self.m.weight.grad, self.reference_m.weight.grad)
+            if self.m.bias is not None:
+                if self.dtype == torch.float16:
+                    torch.testing.assert_close(self.m.bias.grad, self.reference_m.bias.grad, atol=5e-3, rtol=5e-3)
+                else:
+                    torch.testing.assert_close(self.m.bias.grad, self.reference_m.bias.grad)
+
+    def test_sweep(self):
+        for dtype, track_running_stats, channels_last, affine in itertools.product((torch.float, torch.half), (False, True), (False, True), (False, True)):
```

Comment:
could you tell me why this test skips `torch.bfloat16` while the kernel seems to support it https://github.com/NVIDIA/apex/blob/5fa9b1e59b6fbaeabd5e4c5b592df1b6b52ae215/csrc/instance_norm_nvfuser_kernel.cu#L128?

Reply:
Not sure if we need to test/evaluate bf16 by default as it would remove Volta/`sm_70` support? Which I believe was the original use-case/target for instance-norm.

Reply:
I'll look into testing this via https://pytorch.org/docs/stable/generated/torch.cuda.get_device_capability.html

Reply:
Hm, currently eager mode instance_norm supports bfloat16 even on pre-ampere GPUs, so with nvfuser we'd lose that support? We could fall back to old implementation then, I guess.

Reply:
Could just be an artifact of how I'm prototyping things: naively generating bf16 casts and expecting it to work, haven't confirmed that it would be caused by nvfuser.